### PR TITLE
Use `array-bytes` for All Array/Bytes/Hex Operations

### LINKED ISSUE #12111: Using `array-bytes` for Array/Bytes/Hex Operation
Hi! I'm the author of [array-bytes](https://github.com/hack-ink/array-bytes).

This crate is completely optimized for Substrate development.

It's pretty useful and easy to use. And it's completely no-std.

Did you consider replacing `hex` crate, `HexDisplay` struct, and any other `array`/`bytes`/`hex` operation with this crate?

### PR DESCRIPTION
**Pull request by AurevoirXavier**
Close #12111.

cc @bkchr

### CODE REVIEW

**Comment by AurevoirXavier** [comment]
BTW, how do you guys run the tests?

`cargo test --all --all-features` takes 80G RAM. I have to increase my SWAPMEM. The compile progress is so slow.

**Comment by ggwpez** [comment]
> BTW, how do you guys run the tests?
> `cargo test --all --all-features` takes 80G RAM. I have to increase my SWAPMEM. The compile progress is so slow.

As a FRAME developer I normally use the pallet tests to gauge if the CI will likely pass.   
This is often enough since they cover a large part of the code. Together with [cargo-nextest](https://crates.io/crates/cargo-nextest):  
```sh
cargo nextest run -p "pallet-*" --all-features
```

This is no problem on a 32GB machine.

**Comment by AurevoirXavier** [comment]
All tests passed. Please review.

**Review by acatangiu — APPROVED** [review]
all changes look good

**Review by athei — APPROVED** [review]

File: primitives/beefy/src/commitment/test_data.rs [hunk 1]
```diff
@@ -1,536 +1 @@
...
...ce3dc0c33c86e99bcb7816f9ba01558455ad81279df0795cc985580e4fb75d72d948d1107b2ac80a09abed4da8480c746cc321f2319a5e99a830e314d10dd3cd68ce3dc0c33c86e99bcb7816f9ba01558455ad81279df0795cc985580e4fb75d72d948d1107b2ac80a09abed4da8480c746cc321f2319a5e99a830e314d10dd3cd68ce3dc0c33c86e99bcb7816f9ba01558455ad81279df0795cc985580e4fb75d72d948d1107b2ac80a09abed4da8480c746cc321f2319a5e99a830e314d10dd3cd68ce3dc0c33c86e99bcb7816f9ba01558455ad81279df0795cc985580e4fb75d72d948d1107b2ac80a09abed4da8480c746cc321f2319a5e99a830e314d10dd3cd68ce3dc0c33c86e99bcb7816f9ba01558455ad81279df0795cc985580e4fb75d72d948d1107b2ac80a09abed4da8480c746cc321f2319a5e99a830e314d10dd3cd68ce3dc0c33c86e99bcb7816f9ba01558455ad81279df0795cc985580e4fb75d72d948d1107b2ac80a09abed4da8480c746cc321f2319a5e99a830e314d10dd3cd68ce3dc0c33c86e99bcb7816f9ba01558455ad81279df0795cc985580e4fb75d72d948d1107b2ac80a09abed4da8480c746cc321f2319a5e99a830e314d10dd3cd68ce3dc0c33c86e99bcb7816f9ba01558455ad81279df0795cc985580e4fb75d72d948d1107b2ac80a09abed4da8480c746cc321f2319a5e99a830e314d10dd3cd68ce3dc0c33c86e99bcb7816f9ba01558455ad81279df0795cc985580e4fb75d72d948d1107b2ac80a09abed4da8480c746cc321f2319a5e99a830e314d10dd3cd68ce3dc0c33c86e99bcb7816f9ba01";
```
Comment (bkchr) [inline primitives/beefy/src/commitment/test_data.rs:1]: IMO this should be put into a file in `beefy/test-rest/large-raw-commitment` and then being included using `include_bytes!`.

**Review by bkchr — APPROVED** [review]

File: primitives/beefy/src/commitment/test_data.rs (hunk 1 above)
Comment (AurevoirXavier) [reply primitives/beefy/src/commitment/test_data.rs:1]: Done.

Comment (AurevoirXavier) [reply primitives/beefy/src/commitment/test_data.rs]: I follow the spec from `/primitives/trie/test-res/*`.

**Comment by bkchr** [comment]
@AurevoirXavier please merge master

**Comment by shawntabrizi** [comment]
bot rebase

**Comment by athei** [comment]
bot merge

**Pull request merged**